src/main/java/seedu/tinner/model/role/Deadline.java

```diff
@@ -46,6 +46,16 @@ public static boolean isValidDeadline(String test) {
         return true;
     }
 
+    /**
+     * Returns true if the deadline is within a week from the current {@code LocalDateTime}.
+     *
+     * @return true if is within a week from the current date.
+     */
+    public boolean isOneWeekAway() {
+        LocalDateTime oneWeekAway = LocalDateTime.now().plusDays(7);
+        return !this.value.isBefore(LocalDateTime.now()) && oneWeekAway.isAfter(this.value);
+    }
+
```

Comment:
Suggestion that this can be extendable in the future, for example isNthdayAways(N) then user can set the number of days he want like Reminder 7, Reminder 30 etc